### PR TITLE
Standardize formatting

### LINKED ISSUE #13: Standardize formatting
- [x] Add ESLint, `eslint-config-bloq`, Prettier and TS rules.
- [x] Reformat all the code
- [x] Solve any linting errors
- [x] Solve any linting warnings (optional)
- [x] Add `husky`, `lint-staged` and configuration: `npx mrm lint-staged`
- [x] Enable linting on the CI.

Parent: https://github.com/bloqpriv/cloud-product/issues/305

### PR DESCRIPTION
**Pull request by mbonfoster**
We need to standardize code formatting.
This PR adds eslint with bloq's config, Prettier and TS rules.
Also reformats the code to solve errors and warnings, adds husky and lint-staged and enables linting on the ci. 
Closes #13

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -12,4 +12,4 @@ services:
 
 before_script: npm run start:memcached
 
-script: npm run test:e2e
+script: npm run lint && npm run test:e2e
```
Comment (martincarrera) [inline .travis.yml]: I think it's better to run two different scripts here for clarity.

File: package.json [hunk 2]
```diff
@@ -13,19 +14,40 @@
...
+    "husky": "4.2.3",
+    "lint-staged": "10.0.8",
+    "prettier": "1.19.1",
     "chai-as-promised": "7.1.1",
     "mocha": "7.0.1",
     "nyc": "15.0.0",
     "randomstring": "1.1.5",
     "request-promise-native": "1.0.8"
+  },
+  "lint-staged": {
```
Comment (martincarrera) [inline package.json]: We should configure lint-staged in a different file, as in all the other `BloqCloud` repositories. See https://github.com/bloqpriv/cloud-billing/blob/develop/.lintstagedrc.json

**Review by martincarrera — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (gabmontes) [reply package.json]: Like this. We should do the same on `eth-connect`. More reasons towards mono-repos.

File: .travis.yml (hunk 1 above)
Comment (gabmontes) [reply .travis.yml]: IIRC Travis only allows defining one build/test script, hence the need to concatenate commands. Is there any other way?

Comment (martincarrera) [reply .travis.yml]: https://github.com/bloqpriv/cloud-billing/blob/3b3f4133e448d85973dbdf0582c333efac01b8d1/.travis.yml#L19

**Review by martincarrera — DISMISSED** [review]

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+.serverless
```
Comment (gabmontes) [inline .eslintignore]: Serverless is not part of this project. Please remove.

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  "root": true,
+  "rules": {
+    "jsdoc/require-jsdoc": "off",
+    "no-negated-condition": "off",
+    "no-shadow": "warn",
+    "node/no-unpublished-require": "off",
+    "quotes": ["error", "single"],
+    "require-jsdoc": "off",
+    "semi": ["error", "never"]
+  }
```
Comment (gabmontes) [inline .eslintrc.json]: There should be no custom rules except when that makes total sense. Try cleaning this up.

**Review by gabmontes — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 4 above)
Comment (gabmontes) [reply .eslintrc.json]: Should enable `curly: "error"` as mentioned in Slack.

**Review by gabmontes — APPROVED** [review]

**Review by martincarrera — APPROVED** [review]

**Review by jgarzik — APPROVED** [review]
ack

**Pull request merged**